bpf/lib/conntrack.h

```diff
@@ -256,6 +256,11 @@ __ct_lookup(const void *map, struct __ctx_buff *ctx, const void *tuple,
 		} else if (dir == CT_EGRESS) {
 			__sync_fetch_and_add(&entry->tx_packets, 1);
 			__sync_fetch_and_add(&entry->tx_bytes, ctx_full_len(ctx));
+		} else if (dir == CT_SERVICE) {
+#if defined(ENABLE_DSR)	
+			__sync_fetch_and_add(&entry->rx_packets, 1);
+			__sync_fetch_and_add(&entry->rx_bytes, ctx_full_len(ctx));
+#endif
```

Comment:
I'm a bit surprised here. Why can't we just use the `tx_packets` / `tx_bytes` fields for the `CT_SERVICE` case?

Reply:
@julianwiedmann  For DSR, I'd think tx_pacekts/tx_bytes can be used instead of rx_packets and rx_bytes, but is there any case that tx/rx packets tx/rx bytes are both needed?

Reply:
Nope, we shouldn't be doing any `CT_SERVICE` lookup from the reverse direction.

Non-DSR replies get handled by this special-purpose CT entry: https://github.com/cilium/cilium/blob/78a5f177693fb443cd946441f45826bf7fa2437a/bpf/lib/nodeport.h#L2673

Reply:
@julianwiedmann I.C Thanks for point this out. I am new to this and mad a mistake here. It is no need to change the struct ct_entry  here. 

Another issue is I want to expose the metrics like IPVS,  any suggestions for how to expose  them?  
 For more details: 
https://github.com/cilium/cilium/issues/27452
Thanks